### DISCUSSION TITLE
Investigate gi-secret package

### DISCUSSION THREAD

**Discussion by hpfr** [pdobsan/oama#34] [Ideas]
It looks like the [gi-secret](https://hackage.haskell.org/package/gi-secret) package provides libsecret bindings. Leveraging this would make the XDG Secret Service API backend (`ring_lookup`) more robust: no need for users to ensure another package is installed and configure how to execute it.

Opening as an idea rather than an issue because the current implementation does work and because I don’t really know enough Haskell to implement this or determine if the library is any good. Could be an excuse to learn, but no promises :)

**Comment by pdobsan** [comment]
On Sat, Jan 13, 2024 at 09:22:37PM -0800, Liam Hupfer wrote:
> It looks like the
> [gi-secret](https://hackage.haskell.org/package/gi-secret) package
> provides libsecret bindings. Leveraging this would make the XDG Secret
> Service API backend (`ring_lookup`) more robust: no need for users to
> ensure another package is installed and configure how to execute it.

Thank you for the suggestion. Using directly the secretlib bindings has
occurred to me too. I even made repeated efforts to compile the gi
packages and failed every time.

Compiling the gi stuff is notoriously hard in Haskell world.
None of the GHC versions listed in the haskell-gi cabal file's
`tested-with` line even shows up in the current ghcup offering.

So ironically, using secret-tool as an external program is much more
robust than using the secretlib directly. The secret-tool program
usually comes with the secretlib package so there is no extra dependency
involved here.

So, at least for the time being, I'll keep the less elegant but working
solution.

First I'd like to get over with mailctl -> oama transition outlined in
discussion #26. In fact most of it already is done, I have been
successfully using it for a while. Hope to release it soon without
breaking anything.
